eclipse_plugin/src/com/google/googlejavaformat/eclipse/GoogleJavaFormatter.java

```diff
@@ -0,0 +1,114 @@
+import com.google.googlejavaformat.java.Formatter;
+import com.google.googlejavaformat.java.FormatterException;
+import com.google.googlejavaformat.java.Replacement;
+
+/**
+ * Google Java Style Code Formatter.
+ * <p>
+ * Uses <a href= "https://github.com/google/google-java-format">google-java-format</a>, a program
+ * that reformats Java source code to comply with
+ * <a href="https://google.github.io/styleguide/javaguide.html">Google Java Style</a>, to format
+ * Java source code in Eclipse.
+ *
+ * @author Christian Stein
+ * @see <a href= "https://github.com/google/google-java-format">google-java-format</a>
+ */
+public class GoogleJavaFormatter extends CodeFormatter {
+
+  /**
+   * Plug-in ID.
+   */
+  private static final String PLUGIN_ID = GoogleJavaFormatter.class.getCanonicalName();
+
+  /**
+   * The google-java-format formatter instance.
+   */
+  private final Formatter formatter = new Formatter();
+
+  /**
+   * Block indentation: +2 spaces
+   * <p>
+   * Each time a new block or block-like construct is opened, the indent increases by two spaces.
+   * When the block ends, the indent returns to the previous indent level. The indent level applies
+   * to both code and comments throughout the block.
+   *
+   * @see <a href= "https://google.github.io/styleguide/javaguide.html#s4.2-block-indentation">Block
+   *      indentation</a>
+   */
+  @Override
+  public String createIndentationString(int indentationLevel) {
+    return new String(new char[indentationLevel]).replace("\0", "  ");
+  }
+
+  @Override
+  public TextEdit format(
+      int kind, String source, IRegion[] regions, int indentationLevel, String lineSeparator) {
+    List<Range<Integer>> ranges = new ArrayList<>();
+    for (IRegion region : regions) {
+      ranges.add(Range.open(region.getOffset(), region.getOffset() + region.getLength()));
+    }
+    return format(source, ranges);
+  }
+
+  @Override
+  public TextEdit format(
+      int kind, String source, int offset, int length, int indentationLevel, String lineSeparator) {
+    List<Range<Integer>> ranges = Arrays.asList(Range.open(offset, offset + length));
+    return format(source, ranges);
+  }
+
+  protected TextEdit format(String input, List<Range<Integer>> ranges) {
+    TextEdit result = null;
+    try {
+      ImmutableList<Replacement> replacements = formatter.getFormatReplacements(input, ranges);
+      MultiTextEdit multi = new MultiTextEdit();
+      for (Replacement replacement : replacements) {
+        int replaceOffset = replacement.getReplaceRange().lowerEndpoint();
+        int replaceLength = replacement.getReplaceRange().upperEndpoint();
+        String replaceStr = replacement.getReplacementString();
+        String oldString = input.substring(replaceOffset, replaceLength);
+        if (replaceStr.equals(oldString)) {
+          continue;
+        }
+        multi.addChild(new ReplaceEdit(replaceOffset, replaceLength - replaceOffset, replaceStr));
```

Comment:
Feel free to ask myself and @elharo if you have any questions about open
sourcing.

https://github.com/GoogleCloudPlatform/gcloud-eclipse-tools


On Fri, Jun 3, 2016 at 3:51 PM, Christian Stein <notifications@github.com>
wrote:

> In
> eclipse_plugin/src/com/google/googlejavaformat/eclipse/GoogleJavaFormatter.java
> <https://github.com/google/google-java-format/pull/52#discussion_r65765660>
> :
>
> > +  }
> > +
> > +  protected TextEdit format(String input, List<Range<Integer>> ranges) {
> > +    TextEdit result = null;
> > +    try {
> > +      ImmutableList<Replacement> replacements = formatter.getFormatReplacements(input, ranges);
> > +      MultiTextEdit multi = new MultiTextEdit();
> > +      for (Replacement replacement : replacements) {
> > +        int replaceOffset = replacement.getReplaceRange().lowerEndpoint();
> > +        int replaceLength = replacement.getReplaceRange().upperEndpoint();
> > +        String replaceStr = replacement.getReplacementString();
> > +        String oldString = input.substring(replaceOffset, replaceLength);
> > +        if (replaceStr.equals(oldString)) {
> > +          continue;
> > +        }
> > +        multi.addChild(new ReplaceEdit(replaceOffset, replaceLength - replaceOffset, replaceStr));
>
> @lowasser <https://github.com/lowasser> Same here. You can see my debug
> variables... and still I'm not sure, if all cases are met.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/google/google-java-format/pull/52/files/89a31e50354bf736f62925f2aadbb5ded2912735#r65765660>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHf5HZc_TrkFVmjJYaRnqBIxwKb7iI43ks5qIIWmgaJpZM4Intqy>
> .
>